presto-main/src/main/java/com/facebook/presto/tracing/NoopTracerProvider.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.tracing;
+
+import com.facebook.presto.spi.tracing.Tracer;
+import com.facebook.presto.spi.tracing.TracerProvider;
+import com.google.inject.Inject;
+
+public class NoopTracerProvider
+        implements TracerProvider
+{
+    private static final NoopTracer noopTracer = new NoopTracer();
+
+    @Inject
+    public NoopTracerProvider() {}
+
+    @Override
+    public Tracer startTracing(boolean isTracingEnabled)
```

Comment:
- we don't put `public` on interfaces
- why do we need `isTracingEnabled` flag if we are already to start tracing?

Reply:
Yeah it's not beautiful. But we need it so that we can differentiate if user passes this enable_tracing (to be added) session property. Because we cannot use both NoopTracer and ArtilleryTracer at the same time (only one will be binded to Tracer depending on if it is oss or fbinternal)

Reply:
> we don't put public on interfaces
This is a noop implementation. Do we also not put public in noop impls either?

Reply:
How about we call it getTracer() ? 
It is a tracer provider so should make sense.